### PR TITLE
Fix/258 the chatgpt evaluation isnt linked to the correct phase of the explanation

### LINKED ISSUE #258: The ChatGPT evaluation isn't linked to the correct phase of the explanation
## Description

As the image below show, the ChatGPT evaluation is place under the explanation of the first phase. 
But as the content of the evaluation show, it should be place under the explanation of the second phase.

## How to reproduce the bug

1. Play a sequence Face to Face, with Likert and ChatGPT enabled
2. During the Evaluation phase, change the explanation.
3. And finally got to the last phase.
4. As the student that have change their explanation go see the `My result` section.
5. **The error appears**.

## Expected behaviour

The ChatGPT explanation should be placed under the correct explantion version.

## Screenshots (optional)

image

### LINKED ISSUE #260: If an explanation doesn't have an evaluation, ChatGPT won't evaluate it
## Description

If you have an explanation who hasn't been evaluated by a human, Elaastic won't ask ChatGPT to evaluate it.

## How to reproduce the bug

1. Plays a sequence where the number of **explanation to evaluate is 1**. The assignment must have a teacher's explanation. (Execution context : Face to Face, Blended, Distance)
2. Make sure just one student play this sequence
3. As the student play the sequence.
4. During the evaluation phase, you should just be able to evaluate the teacher explanation.
5. **During the last phase you can see that just the teacher have an evaluation (from the student) and the student haven't one.**

## Expected behavior

Every student should have an evaluation from ChatGPT of their explanation.

## Screenshots (optional)

In these picture, we can see a student's explanation without an evaluation.
Image

## Additional Context

This situation always appear when the number of evaluation to make is 1 and their is a teacher's explanation. 

Because the number of evaluation that will be made is equal to the `numberOfSstudent * evaluationToMake`, and the number of explanation to evaluate is equal to `explanationFromStudent + explanationFromTeacher + fakeExplanation`.

So despite the number of student with the extra 1 explanation from the teacher, one explanation won't be evaluate.

### PR DESCRIPTION
**Pull request by Francois389**
Close #258 
Close #260 

# Result

Change the way to visualize evaluation and attempt.

We use a modal to display the `my-result` component (use to display individual result for the learner).

The link to show the modal
image

The modal with the two attempt and an DRAXO evaluation in the second one.
image

## Other change not related to the main US

With this commit 9bcf673417d42db8f4fa0129cbe07af81de9a045 I fix the #260 issue.

***

If the answer didn't have any evaluation istead of diplaying `0 evaluation [...]` we display :
EN
image
FR
image

***

If an answer didn't have a ChatGPT evaluation, we didn't display the link.

No ChaGPT evaluation
image
One ChatGPT evaluation
image

### CODE REVIEW

File: src/main/kotlin/org/elaastic/questions/assignment/sequence/interaction/chatGptEvaluation/ChatGptEvaluationController.kt [hunk 1]
```diff
@@ -107,46 +108,57 @@ class ChatGptEvaluationController(
...
             ) {
-                // If the ChatGPT evaluation is enabled, we add it to the model
-                ChatGptEvaluationModelFactory.build(
-                    evaluation = chatGptEvaluation,
-                    sequence = response.interaction.sequence,
-                    canHideGrading = responseService.canHidePeerGrading(user, response),
-                    responseId = response.id
-                )
-            } else null
+                "You are not authorized to access to these feedback"
```
Comment (FranckSilvestre) [inline src/main/kotlin/org/elaastic/questions/assignment/sequence/interaction/chatGptEvaluation/ChatGptEvaluationController.kt:123]: This message should be localized.

File: src/main/kotlin/org/elaastic/questions/player/PlayerController.kt [hunk 2]
```diff
@@ -696,7 +699,7 @@ class PlayerController(
         val sequence = sequenceService.get(id, true)
         val chatGptEvaluation = chatGptEvaluationService.findEvaluationById(evaluationId)
 
-                // Check authorizations
+        // Check authorizations
         requireAccessThrowDenied(user == chatGptEvaluation!!.response.learner) {
             "You must be the learner of the response to report the evaluation"
```
Comment (FranckSilvestre) [inline src/main/kotlin/org/elaastic/questions/player/PlayerController.kt:704]: This message should be localised.

**Review by FranckSilvestre — CHANGES_REQUESTED** [review]
Only issues concerning some hard coded error message : need for lacalisation.

File: src/main/kotlin/org/elaastic/questions/assignment/sequence/interaction/chatGptEvaluation/ChatGptEvaluationController.kt (hunk 1 above)
Comment (Francois389) [reply src/main/kotlin/org/elaastic/questions/assignment/sequence/interaction/chatGptEvaluation/ChatGptEvaluationController.kt]: fix in d1968c1bbce6662ff9ec7c60168c1f5f8bb5352b

File: src/main/kotlin/org/elaastic/questions/player/PlayerController.kt (hunk 2 above)
Comment (Francois389) [reply src/main/kotlin/org/elaastic/questions/player/PlayerController.kt]: fix in 09e686797934b5b93215be3c23cb797531a07f9e
and  23b2612ed38c667809ebbf82bd3b340eb2fc74e5

**Review by FranckSilvestre — APPROVED** [review]

**Pull request merged**